### DISCUSSION TITLE
Not working in Rust mode

### DISCUSSION THREAD

**Discussion by npajkovsky** [minad/jinx#130] [Q&A] [answered]
Hello,

I'm tying to use jinx in rust-mode/rustic-mode without luck.

The code
```
// Infalid

fn main() {
    // Set reeder.
}
```
It clearly has misspelled words, but does not highlight any.

`M-x jinx-correct-all` says `No misspelled word in whole buffer`. 

Do I have to enable something? 

Note: it works just fine with other modes like c/toml

**Comment by minad** [comment]
Works for me in rust-ts-mode. Misspellings in comments are highlighted properly. I don't have rust-mode and rustic-mode installed. What are they doing differently?

**Reply by npajkovsky** [reply, marked as answer]
Aha, so it works in rust-ts-mode, but not rustic-mode.

**Comment by npajkovsky** [comment]
I know what's going on finally.

```
(use-package tree-sitter
  :straight t
  :diminish tree-sitter-mode
  :config
  ;; activate tree-sitter on any buffer containing code for which it has a parser available
  (global-tree-sitter-mode)
  ;; you can easily see the difference tree-sitter-hl-mode makes for python, ts or tsx
  ;; by switching on and off
  (add-hook 'tree-sitter-after-on-hook #'tree-sitter-hl-mode))
```

The `tree-sitter-hl-mode` must run after the jinx, and therefore no misspellings are shown from jinx.

**Reply by mattiasdrp** [reply]
Is there a way to make jinx work with tree-sitter-hl-mode enabled?
